example/test/function_component_test.dart

```diff
@@ -1,31 +1,50 @@
 import 'dart:html';
 
+import 'package:react/hooks.dart';
 import 'package:react/react.dart' as react;
 import 'package:react/react_dom.dart' as react_dom;
 import 'package:react/react_client.dart';
 
-import 'react_test_components.dart';
+var useStateTestFunctionComponent = react.registerFunctionComponent(UseStateTestComponent, displayName: 'useStateTest');
+
+UseStateTestComponent(Map props) {
+  final count = useState(0);
+
+  return react.div({}, [
+    count.value,
+    react.button({'onClick': (_) => count.set(0)}, ['Reset']),
+    react.button({
+      'onClick': (_) => count.setWithUpdater((prev) {
```

Comment:
#nit it's better to avoid the state change altogether when disabled:
```dart
      'onClick': (_) {
        if (props['enabled']) {
          count.setWithUpdater((prev) => prev + 1);
        }
      },
```